### DISCUSSION TITLE
FirebaseMessaging.deleteToken not deleting token

### DISCUSSION THREAD

**Discussion by dom-the-dev** [capawesome-team/capacitor-firebase#607] [Q&A] [answered]
Hi everyone,

I've been working with the Firebase Messaging plugin and I've encountered a situation where I need to revoke the granted permission by the user.

I've looked through the documentation, but couldn't find any explicit mention of a method to revoke the permission and remove the token. I assumed that deleteToken might serve this purpose, but it doesn't seem to work as expected.

After calling deleteToken, I still receive a token when I subsequently call getToken. Could someone please provide guidance on how to revoke the permission and remove the token properly?

```
    "@capacitor-firebase/messaging": "^5.4.1",
    "@capacitor/app": "5.0.7",
```

Your assistance would be greatly appreciated. Thank you!

**Comment by robingenz** [comment]
Which permission do you want to remove?

**Reply by dom-the-dev** [reply]
`const result = await FirebaseMessaging.requestPermissions();`

User grants permission for recieving push notifications.

**Reply by robingenz** [reply, marked as answer]
No, only the user can revoke the permission via the app settings.
